btcec/btcec_test.go

```diff
@@ -596,12 +596,83 @@ func TestScalarMult(t *testing.T) {
 		exponent.Mul(exponent, new(big.Int).SetBytes(data))
 		xWant, yWant := s256.ScalarBaseMult(exponent.Bytes())
 		if x.Cmp(xWant) != 0 || y.Cmp(yWant) != 0 {
-			t.Fatalf("%d: bad output for %X: got (%X, %X), want (%X, %X)", i, data, x, y, xWant, yWant)
+			t.Fatalf("%d: bad output for %x: got (%x, %x), want (%x, %x)", i, data, x, y, xWant, yWant)
 			break
 		}
 	}
 }
 
+type splitKTest struct {
+	k      string
+	k1, k2 string
+	s1, s2 int
+}
+
+var s256SplitKTests = []splitKTest{
```

Comment:
Please move this inside the `TestSplitK` function so it is consistent with the field tests.

e.g.

```Go
	tests := []struct {
		k      string
		k1, k2 string
		s1, s2 int
	}{
		{
			"...",
		},
	}

	for i, test := range tests {
		// ...
	}
```